daemon/ipcache.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"net"
+
+	"github.com/cilium/cilium/api/v1/models"
+	. "github.com/cilium/cilium/api/v1/server/restapi/policy"
+	"github.com/cilium/cilium/pkg/api"
+	"github.com/cilium/cilium/pkg/identity"
+	"github.com/cilium/cilium/pkg/ipcache"
+
+	"github.com/go-openapi/runtime/middleware"
+)
+
+type getIPCache struct{}
+
+// NewGetIPCacheHandler for the global IP cache
+func NewGetIPCacheHandler() GetIPCacheHandler {
+	return &getIPCache{}
+}
+
+func (h *getIPCache) Handle(params GetIPCacheParams) middleware.Responder {
+	listener := &ipCacheDumpListener{}
+	if params.Cidr != nil {
+		_, cidrFilter, err := net.ParseCIDR(*params.Cidr)
+		if err != nil {
+			return api.Error(GetIPCacheBadRequestCode, err)
+		}
+		listener.cidrFilter = cidrFilter
+	}
+	ipcache.IPIdentityCache.RLock()
+	ipcache.IPIdentityCache.DumpToListenerLocked(listener)
+	ipcache.IPIdentityCache.RUnlock()
+	if len(listener.entries) == 0 {
+		return NewGetIPCacheNotFound()
+	}
+
+	return NewGetIPCacheOK().WithPayload(&models.IPCache{
```

Comment:
The code here doesn't seem to match up with the API defined. It seems to be fixed in the next commit though. Can you fix it up so we don't break bisecting?

Reply:
Good catch, that was not my intention! Apparently I did a sloppy job of rewriting the git history :disappointed_relieved: